VARIABLES.md

```diff
@@ -67,6 +67,7 @@
 | s3_inventory_format | Output format for S3 inventory results. Can be Parquet, ORC, CSV | string | `ORC` | no |
 | s3_inventory_update_schedule | Cron schedule to update S3 inventory tables (if enabled). Defaults to every 12 hours. | string | `0 */12 * * *` | no |
 | s3_lifecycle_policy_transition_period | Number of days for transition to a different storage class using lifecycle policy. | string  | `30` | no |
+| s3_object_expiration_days | Number of days after which objects in the Apiary S3 buckets expire | number | `null` | no
```

Comment:
If my understanding is correct, we want to configure this for certain buckets only (things like analyst sandbox, scratch areas, etc).  But from this line where it is added as a module variable, and then in `s3.tf` where it is used as the default value if `s3_object_expiration_days` is *not* configured in the `apiary_managed_schemas` list of maps, it seems like the user of this module could end up configuring a value for this variable that could wipe out all their bucket objects inadvertently after the timer expires.

I am understanding this correctly?  If so, I strongly suggest we do NOT have an overall default value that is applied if the setting is not set on a per-bucket basis.  Was an overall default one of the requirements for this feature?

Reply:
Also, the above line documents the overall default variable `s3_object_expiration_days`.  But we also need documentation about setting it per-bucket.  Could you update the documentation for `apiary_managed_schemas` to expand it to documenting *ALL* fields that are allowed in each map entry of `apiary_managed_schemas` (see documentation for `apiary_assume_roles` in this same file for an example).  Please add a section to `VARIABLES.md` that completely documents `apiary_managed_schemas`.  Yes, I know adding all the documentation is outside the scope of adding this one feature, but the longer we neglect it, the more it gets impossible to understand what `apiary_managed_schemas` does.